### PR TITLE
Outbound retransmissions rebase

### PR DESCRIPTION
**Pull request by bblaszkow06**
Migrates PoC of outbound retransmissions to the current plugin version

### CODE REVIEW

File: lib/membrane/rtp/outbound_rtx_controller.ex [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      )
+    end
+
+    {[buffer: {:output, buffers_to_retransmit}], %{state | store: store}}
+  end
+
+  @impl true
+  def handle_event(pad, event, ctx, state), do: super(pad, event, ctx, state)
+
+  defp sn_to_index(seq_num), do: rem(seq_num, @max_store_size)
```
Comment (daniel-jodlos) [inline lib/membrane/rtp/outbound_rtx_controller.ex:72]: Can we use something a little more verbose than `sn` here? At least
```suggestion
  defp seq_num_to_index(seq_num), do: rem(seq_num, @max_store_size)
```
or maybe even
```suggestion
  defp sequence_number_to_index(seq_num), do: rem(seq_num, @max_store_size)
```

File: lib/membrane/rtp/outbound_rtx_controller.ex [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    if buffer != nil and buffer.metadata.rtp.sequence_number == seq_num and
+         time_elapsed?(last_rtx_time, now) do
+      store = Map.put(store, idx, {now, buffer})
+      {buffer, store}
+    else
+      {nil, store}
+    end
+  end
+
+  defp time_elapsed?(nil, _now), do: true
```
Comment (daniel-jodlos) [inline lib/membrane/rtp/outbound_rtx_controller.ex:87]: This function name is also a little weird. You have to look at the implementation to know what it does, so maybe:
```suggestion
  defp min_rtx_interval_elapsed?(nil, _now), do: true
```

**Review by daniel-jodlos — APPROVED** [review]
approval-0

**Comment by bblaszkow06** [comment]
Automerge prevented applying the fixes, will do in another PR

**Pull request merged**